tools/alpine/go-compile.sh

```diff
@@ -13,6 +13,9 @@ dir="$1"
 
 cd "$dir"
 
+# Use '-mod=vendor' for builds which have switched to go modules
+[ -f go.mod -a -d vendor ] && export GOFLAGS="-mod=vendor"
```

Comment:
nit: this is true for builds that have switches to go modules AND use `go mod vendor`  to populate a vendor directory.

It's a little bothersome to me me that we have a script here and also one in `pkg/go-compile`.
It might be nice to deprecate this one and just have the latter.

Reply:
Agreed. At this point, though, it was _so_ much effort over several weeks just to get everything working, I wasn't going to try refactoring that.

Similar nit: only two packages downstream use containerd; so why have it in alpine, where the entire base changes if we bump alpine version? I plan on breaking it out... _after_ this is in

Reply:
changing this would require a **full** day of work, as the `go-compile.sh` is in alpine, which means changing everything downstream, a long slow process.